amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/domain/OasLikeOperationParser.scala

```diff
@@ -119,19 +115,21 @@ abstract class OasOperationParser(entry: YMapEntry, adopt: Operation => Operatio
           .filter(y => !isOasAnnotation(y.key.as[YScalar].text))
           .foreach {
             entry =>
-              val node = ScalarNode(entry.key).text()
+              val node = ScalarNode(entry.key)
               responses += OasResponseParser(
-                entry.value.as[YMap], { r =>
-                  r.set(ResponseModel.Name, node, Annotations(entry.key))
-                    .adopted(operation.id)
-                    .withStatusCode(r.name.value())
-                  r.annotations ++= Annotations(entry)
-                  // Validation for OAS 3
-                  if (ctx.isInstanceOf[Oas3WebApiContext] && entry.key.tagType != YType.Str)
-                    ctx.eh.violation(InvalidStatusCode,
-                                     r.id,
-                                     "Status code for a Response object must be a string",
-                                     entry.key)
+                entry.value.as[YMap], {
+                  r =>
+                    r.withName(node)
+                      .adopted(operation.id)
+                      .set(ResponseModel.StatusCode, node.text(), Annotations.inferred())
+                    if (!r.annotations.contains(classOf[SourceAST]))
```

Comment:
Why ask instead of overwriting?

Reply:
because there is no easy way to override